analysis/scripts/2233-power-calculations.Rmd

```diff
@@ -257,105 +278,153 @@ This functions calculates the power of different average treatment effects for s
-              in_ate = in_ate,
-              in_population_mean = in_population_mean , #population_mean_color_contrast_issues,
-              in_prop_assigned_treatment = in_prop_assigned_treatment #population_treatement_proportion,
-              )
-  )
-  
-  out <- 
-    sims %>% 
-  filter(term=="treatment") %>%
-  group_by(regression) %>%
-  summarise(beta_mean = mean(estimate),
-            beta_sd = sd(estimate),
-            power = mean(p.value <= 0.05)) %>% 
-    ungroup() %>% 
-    mutate(ate = in_ate)
-  
-  return(out)
+    sims <-
+        1:in_iterations %>%
+        map_dfr(~simulation_01(in_iteration = .,
+                in_population_size = in_population_size,
+                in_ate = in_ate,
+                in_population_mean = in_population_mean ,
+                in_prop_assigned_treatment = in_prop_assigned_treatment
+    )
+        )
+
+        out <-
+            sims %>%
+            filter(term=="treatment") %>%
+            group_by(regression) %>%
+            summarise(beta_mean = mean(estimate),
+                beta_sd = sd(estimate),
+                mean_est_se = mean(std.error),
+                power = mean(p.value <= 0.05),
+                coverage = mean( true_ate >= conf.low & true_ate <= conf.high),
+                bias = mean(estimate - true_ate),
+                mean_ate = mean(true_ate)) %>%
+            ungroup() %>%
+            mutate(ate = in_ate)
+
+        return(out)
 }
 ```
 
+```{r testing_fns, eval=FALSE, include=FALSE}
+## This chunk is just to allow testing of the functions.
+##
+##
+set.seed(12345)
+sim_out <- simulation_01(in_iteration = 1,
+    in_population_size = 1000,
+    in_ate = params$ate_min,
+    in_population_mean = 2,
+    in_prop_assigned_treatment = params$treatment_probability)
+sim_out
+
+## When in_ate=0 power should be the same as false positive rate
+## and so should be .05 or less (plus or minus simulation error). Coverage should
+## be at least .95.  We also want beta_sd to be very close to est_se if our
+## estimated standard errors are approximating the true standard error well
+
+power_out <- power_01(in_iterations=1000,
+    in_population_size=1000,
+    in_ate=0,
+    in_population_mean=2,
+    in_prop_assigned_treatment= params$treatment_probability)
+
+power_out
+
+## We expect more than .05 power but coverage should stay close to .95 or above.
+power_out_ate1 <- power_01(in_iterations=1000,
+    in_population_size=1000,
+    in_ate=-1,
+    in_population_mean=2,
+    in_prop_assigned_treatment= params$treatment_probability)
+
+power_out_ate1
+```
+
 ### Calculating power
 
 We calculate the power for treatment effects between `r params$ate_min` and `r params$ate_max` incremented by `r params$ate_increment` for `r scales::comma(params$simulations)` of this function.
 
 ```{r}
+## Trying to speed this up with a parallel version of map_dfr
+plan(multisession,workers=availableCores()-1)
```

Comment:
@bpffjl  Not sure if this will work on whatever systems you are using. But I just wanted to speed things up.

Reply:
Great. I haven't tried this in the VDI before. Finger crossed.